autotest/gcore/mask.py

```diff
@@ -1137,6 +1137,50 @@ def mask_25():
 
     return 'success'
 
+###############################################################################
+# Test on a GDT_UInt16 1band data
+
+
+def mask_26():
+
+    ds = gdal.GetDriverByName('GTiff').Create('/vsimem/mask_26.tif', 100, 100, 2,
+                                              gdal.GDT_UInt16, options=['ALPHA=YES'])
+    ds.GetRasterBand(1).Fill(65565)
+    ds.GetRasterBand(2).Fill(65565)
+
+    if ds.GetRasterBand(1).GetMaskFlags() != gdal.GMF_ALPHA + gdal.GMF_PER_DATASET:
+        gdaltest.post_reason('Did not get expected mask.')
+        return 'fail'
+    mask = ds.GetRasterBand(1).GetMaskBand()
+
+    # IRasterIO() optimized case
+    import struct
+    if struct.unpack('B', mask.ReadRaster(0, 0, 1, 1))[0] != 255:
+        gdaltest.post_reason('fail')
+        print(struct.unpack('B', mask.ReadRaster(0, 0, 1, 1))[0])
+        return 'fail'
+
+    # IReadBlock() code path
```

Comment:
you could remove that block and the following one, as they test the same code paths as the RGBA case